chalk-parse/src/ast.rs

```diff
@@ -290,3 +292,47 @@ pub enum Goal {
     // Additional kinds of goals:
     Leaf(LeafGoal),
 }
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+/// This allows users to add arbitrary `A :- B` clauses into the
+/// logic; it has no equivalent in Rust, but it's useful for testing.
```

Comment:
This comment is surely wrong

Reply:
But I'm not convinced there's a role for this struct regardless -- I guess the intention was to give some way to "declare" scalar types? I think we should just build them in.